src/pages/index.js

```diff
@@ -18,24 +20,50 @@ const profileEditForm = document.forms["profile-form"];
 const cardListElement = document.querySelector(".cards__list");
 const addNewCardBtn = document.querySelector(".profile__add-button");
 const newCardForm = document.forms["card-form"];
-// const profileEditModal = document.querySelector("#profile-edit-modal");
-// const profileTitle = document.querySelector(".profile__title");
-// const profileDescription = document.querySelector(".profile__description");
-// const profileTitleInput = document.querySelector("#modal-title-input");
-// const profileDescriptionInput = document.querySelector(
-//   "#modal-description-input"
-// );
-// const cardModal = document.querySelector("#add-card-modal");
-// const cardTitleInput = newCardForm.querySelector("#modal-card-title");
-// const cardLinkInput = newCardForm.querySelector("#modal-card-link");
-// const imagePreviewModal = document.querySelector("#modal-image-preview");
-// const modalImage = imagePreviewModal.querySelector(".modal__image-preview");
-// const modalTitle = imagePreviewModal.querySelector(".modal__image-title");
-// const popups = document.querySelectorAll(".modal");
+const deleteButton = document.querySelector(".modal__button-delete");
+const avatarEditButton = document.querySelector(".profile__image-overlay");
+const avatarForm = document.forms["avatar-form"];
+
+/*=============================================
+=                     API                     =
+=============================================*/
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "abab587d-903d-4fa7-90c2-ee61a34ba949",
+    "Content-Type": "application/json",
+  },
+});
+
+/*=============================================
+=            Render & Delete Cards            =
+=============================================*/
+// initalizate delete card button before new section
+const deleteCardPopup = new PopupConfirmation(
+  "#delete-card-modal",
+  handleDeleteCard
+);
+deleteCardPopup.setEventListeners();
+
+// handle the delete button click
+function handleDeleteCard(cardElement) {
+  const cardId = cardElement.dataset.id;
+
+  // delete request to the API
+  api.deleteCard(cardId).then(() => {
+    cardElement.remove(); // remove card from the DOM
+  });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**